### PR TITLE
fixing unbound local variable on search dependencies

### PR DESCRIPTION
**Pull request by kobymeir**
## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6926

## Description
Fixes unbound local variable on search dependencies XSIAM build.

### CODE REVIEW

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -134,45 +134,31 @@ def get_pack_dependencies(client: demisto_client, pack_data: dict, lock: Lock):
...
+            response_type='object'
+        )
         if 200 <= status_code < 300:
             dependencies_data: list = []
             dependants_ids = [pack_id]
             response_data = response_data.get('dependencies', [])
             create_dependencies_data_structure(response_data, dependants_ids, dependencies_data, dependants_ids)
-            dependencies_str = ', '.join([dep['id'] for dep in dependencies_data])
             if dependencies_data:
+                dependencies_str = ', '.join([dep['id'] for dep in dependencies_data])
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:152]: Good catch! :-)

File: Tests/Marketplace/search_and_install_packs.py [hunk 2]
```diff
@@ -134,45 +134,31 @@ def get_pack_dependencies(client: demisto_client, pack_data: dict, lock: Lock):
...
                 logging.debug(f'Found the following dependencies for pack {pack_id}: {dependencies_str}')
             return dependencies_data
         if status_code == 400:
             logging.error(f'Unable to find dependencies for {pack_id}.')
             return []
-        else:
-            msg = response_data.get('message', '')
-            raise Exception(f'Failed to get pack {pack_id} dependencies - with status code {status_code}\n{msg}\n')
-    except Exception:
-        logging.exception(f'The request to get pack {pack_id} dependencies has failed. {status_code=}.')
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:175]: Add `msg` in the log as well.

**Review by tkatzir — APPROVED** [review]
Approved, with some comments.

File: Tests/Marketplace/search_and_install_packs.py (hunk 2 above)
Comment (kobymeir) [reply Tests/Marketplace/search_and_install_packs.py:175]: there is no need, it will be added as the `{ex}` in the f-string, when the exception is caught.

**Comment by kobymeir** [comment]
pre-commit - error on code that wasn't changed by me.
code.pan.run - failed on unrelated timeout for upload.

**Pull request merged**